### ISSUE TITLE
Formatting with -A breaks the code

### ISSUE THREAD

**Issue by saks** [rubocop/rubocop#14120]
Automatic formatting break the code starting from rubocop version 1.75.3.

--------

## Expected behavior

Ruby code can still be executed/interpreted after running `rubocop -A file.rb`

## Actual behavior

Ruby fails to parse code after formatting.

## Steps to reproduce the problem

Example code (content of `/tmp/1.rb`):
```ruby
# frozen_string_literal: true

method_a(1, 2, (method_b 3))
```

Output:
```
➜  ~ rubocop -A --debug /tmp/1.rb
For /Users/saksmlz: Default configuration from /Users/saksmlz/.gem/ruby/3.4.0/gems/rubocop-1.75.3/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /tmp/1.rb
F

Offenses:

/tmp/1.rb:3:16: C: [Corrected] Style/RedundantParentheses: Don't use parentheses around a method argument.
method_a(1, 2, (method_b 3))
               ^^^^^^^^^^^^
/tmp/1.rb:3:25: F: Lint/Syntax: unexpected token tINTEGER
(Using Ruby 2.7 parser; configure using TargetRubyVersion parameter, under AllCops)
method_a(1, 2, method_b 3)
                        ^

1 file inspected, 2 offenses detected, 1 offense corrected
```

File content after formatting:
```ruby
# frozen_string_literal: true

method_a(1, 2, method_b 3)
```

Error:
```
➜  ~ ruby /tmp/1.rb
/tmp/1.rb: --> /tmp/1.rb

unexpected integer; expected a `)` to close the arguments
unexpected integer, expecting end-of-input
unexpected ')', expecting end-of-input
unexpected ')', ignoring it

> 3  method_a(1, 2, method_b 3)

/tmp/1.rb:3: syntax errors found (SyntaxError)
  1 | # frozen_string_literal: true
  2 |
> 3 | method_a(1, 2, method_b 3)
    |                         ^ unexpected integer, expecting end-of-input
    |                         ^ unexpected integer; expected a `)` to close the arguments
    |                          ^ unexpected ')', ignoring it
    |                          ^ unexpected ')', expecting end-of-input
```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.75.3 (using Parser 3.3.8.0, rubocop-ast 1.44.1, analyzing as Ruby 2.7, running on ruby 3.4.3) [arm64-darwin24]
```
